e2e-tests/testdata/echo_client/client.js

```diff
@@ -0,0 +1,912 @@
+// Code generated by the Encore devel client generator. DO NOT EDIT.
+
+
+/**
+ * Local is the base URL for calling the Encore application's API.
+ */
+export const Local = "http://localhost:4000"
+
+/**
+ * Environment returns a BaseURL for calling the cloud environment with the given name.
+ */
+export function Environment(name) {
+    return `https://${name}-slug.encr.app`
+}
+
+/**
+ * Client is an API client for the slug Encore application. 
+ */
+export default class Client {
+    cache
+    di
+    echo
+    endtoend
+    flakey_di
+    middleware
+    test
+    validation
+
+
+    /**
+     * Creates a Client for calling the public and authenticated APIs of your Encore application.
+     *
+     * @param target  The target which the client should be configured to use. See Local and Environment for options.
+     * @param options Options for the client
+     */
+    constructor(target = "prod", options) {
```

Comment:
Having a default value followed by something without defaults looks quite strange. How about:
```suggestion
    constructor(target = "prod", options = undefined) {
```